### PR TITLE
Added solution to HW26-1

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

**Review by TimoxaV — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/cart/AddToCartController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddToCartController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/cart/AddToCartController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/cart/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> addedProducts = shoppingCartService.getByUserId(USER_ID).getProducts();
+
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/cart/ShoppingCartController.java]: let's remove empty line

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
+
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/product/AddProductController.java]: empty line)

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        Product newProduct = new Product(name, Double.parseDouble(price));
+        productService.create(newProduct);
+        resp.sendRedirect(req.getContextPath() + "/");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/product/AddProductController.java]: maybe better redirect to the list of all products? it is optional)

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -11,6 +11,86 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>injectData</servlet-name>
+        <servlet-class>com.internet.shop.controllers.InjectDataController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>injectData</servlet-name>
+        <url-pattern>/injectData</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java (hunk 4 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java]: totally agree

File: src/main/webapp/web.xml (hunk 5 above)
Comment (MichaelShevchenko) [reply src/main/webapp/web.xml]: my bad

File: src/main/java/com/internet/shop/controllers/cart/AddToCartController.java (hunk 1 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/controllers/cart/AddToCartController.java]: done

File: src/main/java/com/internet/shop/controllers/cart/ShoppingCartController.java (hunk 2 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/controllers/cart/ShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java (hunk 3 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java]: done

File: src/main/webapp/web.xml (hunk 5 above)
Comment (MichaelShevchenko) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java (hunk 4 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**